api/controller.py

```diff
@@ -416,36 +399,21 @@ def circulation(self):
         library_id = flask.request.library.id
         return self.manager.circulation_apis[library_id]
     
-    def load_lane(self, language_key, name):
+    def load_lane(self, lane_identifier):
         """Turn user input into a Lane object."""
         library_id = flask.request.library.id
         top_level_lane = self.manager.top_level_lanes[library_id]
 
-        if language_key is None and name is None:
+        if lane_identifier is None:
             return top_level_lane
 
-        lanelist = top_level_lane.sublanes
-        if not language_key in lanelist.by_languages:
-            return NO_SUCH_LANE.detailed(
-                _("Unrecognized language key: %(language_key)s", language_key=language_key)
-            )
-
-        if name:
-            name = name.replace("__", "/")
-
-        lanes = lanelist.by_languages[language_key]
-
-        if not name:
-            defaults = [x for x in lanes.values() if x.default_for_language]
-            if len(defaults) == 1:
-                # This language has one, and only one, default lane.
-                return defaults[0]
+        lane = get_one(self._db, Lane, identifier=lane_identifier, library_id=library_id)
```

Comment:
This can wait for a future branch, but you have a great opportunity to turn Lane into a HasFullTableCache, with (lane_identifier, library_id) as the cache key.